### PR TITLE
Allow _.bindAll to take arrays

### LINKED ISSUE #1996: can _.bindAll accept an array of function names
I wrote this Stackoverflow issue for this common (imo) Underscore + Backbone pattern. 
http://stackoverflow.com/questions/27697299/passing-an-array-of-names-to-bindall-arguments-list-has-wrong-type/

The best answer was to do this:
```
_.bindAll.apply(_, [this].concat(_.values(this.events)));
```
Would be much more readable if it could be written like this:
```
_.bindAll(_.values(this.events));
```
Many of the Underscore functions accept both a string or an array of strings, why not this one?

### PR DESCRIPTION
**Pull request by jridgewell**
Closes #1996, supercedes #2005.

### CODE REVIEW

**Comment by akre54** [comment]
Remind me the advantage of this over varargs-style passing? What's the real-world use case? (https://github.com/jashkenas/underscore/issues/1996#issuecomment-68912000)

**Comment by jridgewell** [comment]
I see the biggest win as parity with the other object functions `_.pick` and `_.omit` (I've always considered  `_.bindAll` an object function).

File: underscore.js [hunk 1]
```diff
@@ -716,10 +716,11 @@
   // are the method names to be bound. Useful for ensuring that all callbacks
   // defined on an object belong to it.
   _.bindAll = function(obj) {
-    var i, length = arguments.length, key;
-    if (length <= 1) throw new Error('bindAll must be passed function names');
-    for (i = 1; i < length; i++) {
-      key = arguments[i];
+    var keys = flatten(arguments, false, false, 1);
```
Comment (megawac) [inline underscore.js]: With #2140 

```js
_.bindAll(obj, params) {
   params = flatten(params);
}
```

**Comment by jashkenas** [comment]
Rebase for a clean merge?

**Comment by jridgewell** [comment]
Rebased.

**Pull request merged**